src/common/utils/fqdn.cpp

```diff
@@ -23,36 +23,51 @@
 #include "logger.hpp"
 #include <boost/asio.hpp>
 #include <iostream>
+#include <chrono>
+#include <thread>
+
+#define MAX_NB_RESOLVE_TRIES 4
+#define TIME_BETWEEN_TRIES 2
 
 bool fqdn::resolve(
     const std::string& host_name, std::string& address, uint32_t& port,
     uint8_t& addr_type, const std::string& protocol) {
-  try {
-    boost::asio::io_context io_context = {};
+  int tries = 0;
+  Logger::spgwu_app().debug("Resolving a DNS (name %s)", host_name.c_str());
+  while (tries < MAX_NB_RESOLVE_TRIES) {
+    try {
+      boost::asio::io_context io_context = {};
+      Logger::spgwu_app().debug("Resolving DNS Try #%u", tries);
 
-    boost::asio::ip::tcp::resolver resolver{io_context};
-    boost::asio::ip::tcp::resolver::results_type endpoints =
-        resolver.resolve(host_name, protocol);
+      boost::asio::ip::tcp::resolver resolver{io_context};
+      boost::asio::ip::tcp::resolver::results_type endpoints =
+          resolver.resolve(host_name, protocol);
 
-    addr_type = 0;  // IPv4 by default
-    for (auto it = endpoints.cbegin(); it != endpoints.cend(); it++) {
-      // get the first Endpoint
-      boost::asio::ip::tcp::endpoint endpoint = *it;
-      address                                 = endpoint.address().to_string();
-      port                                    = endpoint.port();
-      Logger::spgwu_app().debug(
-          "Resolve a DNS (name %s, protocol %s): Ip Addr %s, port %u",
-          host_name.c_str(), protocol.c_str(), address.c_str(), port);
-      if (endpoint.address().is_v4())
-        addr_type = 0;
-      else
-        addr_type = 1;
-      return true;
+      addr_type = 0;  // IPv4 by default
+      for (auto it = endpoints.cbegin(); it != endpoints.cend(); it++) {
+        // get the first Endpoint
+        boost::asio::ip::tcp::endpoint endpoint = *it;
+        address = endpoint.address().to_string();
+        port    = endpoint.port();
+        Logger::spgwu_app().debug(
+            "Resolved a DNS (name %s, protocol %s): Ip Addr %s, port %u",
+            host_name.c_str(), protocol.c_str(), address.c_str(), port);
+        if (endpoint.address().is_v4())
+          addr_type = 0;
```

Comment:
I would recommend the use of kConstants here instead of 0,1, or use boolean isV4, etc

Reply:
I did not touch that part.
This is the original code.